lib/excon/socket.rb

```diff
@@ -241,6 +241,14 @@ def write_nonblock(data)
           # I wish that this API accepted a start position, then we wouldn't
           # have to slice data when there is a short write.
           written = @socket.write_nonblock(data)
+        rescue Errno::EFAULT
+          if OpenSSL::OPENSSL_LIBRARY_VERSION.split(' ')[1] == '1.0.2'
+            msg = "The version of OpenSSL this ruby is built against (1.0.2) has a vulnerability
+                   which causes a fault. For more, see https://github.com/excon/excon/issues/467"
+            raise SecurityError.new(msg)
```

Comment:
I hadn't seen `SecurityError` before. I can find it in new Ruby docs, but how far back is it implemented? I just don't want to have a backwards incompatible thing, so StandardError or something might be safer (unless we know for sure SecurityError has been around for a while, definitely possible I just haven't run across it before).